iree/compiler/Dialect/HAL/Target/LLVM/LLVMIRPasses.cpp

```diff
@@ -66,6 +66,8 @@ std::unique_ptr<llvm::TargetMachine> createTargetMachine(
 LogicalResult runLLVMIRPasses(const LLVMTargetOptions &options,
                               llvm::TargetMachine *machine,
                               llvm::Module *module) {
+  if(options.addressSanitizer)
+    assert(false && "Address Sanitization feature has not been implemented yet.");
```

Comment:
avoid assert for expected conditions - it is generally not useful (a lot of people build and run in release mode with asserts disabled) and really difficult to debug. MLIR has `emitError`/`return failure()` to gracefully handle this with good error messages.